src/Validation/gen/Parsers/ValidationsGenerator.TypesParser.cs

```diff
@@ -105,6 +105,11 @@ internal bool TryExtractValidatableType(ITypeSymbol incomingTypeSymbol, WellKnow
         return true;
     }
 
+    private static bool IsEqualityContract(IPropertySymbol prop) =>
+        prop.Name == "EqualityContract"
+        && prop.Type.ToDisplayString() == "System.Type"
```

Comment:
Add System.Type to https://github.com/dotnet/aspnetcore/blob/main/src/Shared/RoslynUtils/WellKnownTypeData.cs then use WellKnownTypes to do the type comparison here.